src/main/java/slimeknights/tconstruct/library/client/RenderUtil.java

```diff
@@ -446,12 +459,22 @@ public static int blue(int c) {
    * @param text  Text to add information to.
    */
   public static void liquidToString(FluidStack fluid, List<String> text) {
-    int amount = fluid.amount;
-    if(smelteryLoaded && !Util.isShiftKeyDown()) {
-      List<FluidGuiEntry> entries = fluidGui.get(fluid.getFluid());
+    liquidToString(fluid.getFluid(), fluid.amount, text);
+  }
+
+  /**
+   * Adds information for the tooltip based on the fluid stacks size.
+   *
+   * @param fluid  Input fluid, only used for type calculation
+   * @param amount Fluid amount
+   * @param text   Text to add information to.
+   */
+  public static void liquidToString(@Nullable Fluid fluid, int amount, List<String> text) {
```

Comment:
Wrong kind of extraction. You never want to pass null into this function. Extract the part you're actually interested in to be reused.